compiler/record-minmax/driver/Driver.cpp

```diff
@@ -113,17 +119,35 @@ int entry(const int argc, char **argv)
 
   RecordMinMax rmm;
 
+  // TODO: support parallel record for profile with random data
+  if (parallel_record and not arser["--input_data"])
+  {
+    throw std::runtime_error("Parallel recording is used only for h5 now\n");
+  }
+
   // Initialize interpreter and observer
   rmm.initialize(input_model_path);
 
   if (arser["--input_data"])
   {
     auto input_data_path = arser.get<std::string>("--input_data");
 
+    // TODO: support parallel record from file and dir input data format
+    if (parallel_record and not(input_data_format == "h5") and not(input_data_format == "hdf5"))
+    {
+      throw std::runtime_error("Parallel recording is used only for h5 now\n");
+    }
+
     if (input_data_format == "h5" || input_data_format == "hdf5")
     {
       // Profile min/max while executing the H5 data
-      rmm.profileData(mode, input_data_path, min_percentile, max_percentile);
+      if (not parallel_record)
+        rmm.profileData(mode, input_data_path, min_percentile, max_percentile);
+      else
+      {
+        std::cout << "Using parallel recording" << std::endl;
```

Comment:
This log will be printed even when `_OPENMP` is not defined but a user gives `--parallel true`.

I think warning message should be printed at that case, e.g., `"Parallel recording is not available in this platform. A single thread is used for recording."`